docker.go

```diff
@@ -709,24 +729,31 @@ func (c *DockerContainer) StartLogProducer(ctx context.Context) error {
 				}
 			}
 		}
-	}(c.stopProducer, c.producerDone)
+	}(c.stopProducer, c.producerDone, c.producerError)
 
 	return nil
 }
 
 // StopLogProducer will stop the concurrent process that is reading logs
 // and sending them to each added LogConsumer
 func (c *DockerContainer) StopLogProducer() error {
+	c.producerMutex.Lock()
+	defer c.producerMutex.Unlock()
 	if c.stopProducer != nil {
 		c.stopProducer <- true
 		// block until the producer is actually done in order to avoid strange races
 		<-c.producerDone
 		c.stopProducer = nil
 		c.producerDone = nil
+		return <-c.producerError
 	}
 	return nil
 }
 
+func (c *DockerContainer) GetLogProducerErrorChannel() <-chan error {
```

Comment:
Well, that's the only way for the consumer to be able to listen to errors and react to them. I can point you to our code where we use it, although it might not be the clearest example as there we retry starting/listening in a detached goroutine. 

When I have a second I'd like to use Mockery or something similar to mock
```
r, err := c.provider.client.ContainerLogs(ctx, c.GetContainerID(), options)
```

so that we can cause timeouts or other read errors to make sure the channel works as expected and provide an example of how to use it. WDYT?

Reply:
You already have a `c.Logs(ctx)` method in the container interface. Probably not the simplest thing to have your own mock for that, but it could work

Reply:
It's the other way around: `c.Logs` calls `c.provider.client.ContainerLogs`: https://github.com/testcontainers/testcontainers-go/blob/9c97abfa233364d792ff7723606bebc242237c19/docker.go#L323